<?xml version="1.0"?>
<launch>
    <!-- MAVROS posix SITL environment launch script -->
    <!-- launches Gazebo environment and 2x: MAVROS, PX4 SITL, and spawns vehicle -->
    <!-- vehicle model and world -->
    <arg name="est" default="ekf2"/>
    <arg name="vehicle" default="typhoon_h480"/>
    <!--<arg name="world" default="$(find mavlink_sitl_gazebo)/worlds/four_big_doors_with_ground.world"/>-->
    <arg name="world" default="$(find mavlink_sitl_gazebo)/worlds/visual_formation_sim1.world"/>
    <!-- gazebo configs -->
    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>
    <!-- Gazebo sim -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="gui" value="$(arg gui)"/>
        <arg name="world_name" value="$(arg world)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="verbose" value="$(arg verbose)"/>
        <arg name="paused" value="$(arg paused)"/>
    </include>
    <!-- UAV1 -->
    <group ns="uav1">
        <!-- MAVROS and vehicle configs -->
        <arg name="ID" value="1"/>
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn_sdf.launch">
            <arg name="x" value="0"/>
            <arg name="y" value="0"/>
            <arg name="z" value="0"/>
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="0"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14561"/>
            <arg name="mavlink_tcp_port" value="4561"/>
            <arg name="ID" value="$(arg ID)"/>
        </include>
        <!-- MAVROS -->
        <!--<arg name="fcu_url" default="udp://:14541@localhost:14581"/>
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <arg name="tgt_component" value="1"/>
        </include>-->
        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find px4_application)/config/mavros_config/px4_pluginlists.yaml" />
            <arg name="config_yaml" value="$(find px4_application)/config/mavros_config/px4_config_1.yaml" />
            <arg name="fcu_url" value="udp://:14541@localhost:14581" />
            <arg name="gcs_url" value="" />
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
            <arg name="tgt_component" value="1" />
	    </include>

    <!-- <node pkg="tf" type="static_transform_publisher" name="tf_depth_camera"
            args="0 0 0 -1.57 0 -1.57 base_link iris_1/depth_camera_link 100"/> -->
    </group>
    <!-- UAV2 -->
    <group ns="uav2">
        <!-- MAVROS and vehicle configs -->
        <arg name="ID" value="2"/>
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn_sdf.launch">
            <arg name="x" value="-2"/>
            <arg name="y" value="0"/>
            <arg name="z" value="0"/>
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="0"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14562"/>
            <arg name="mavlink_tcp_port" value="4562"/>
            <arg name="ID" value="$(arg ID)"/>
        </include>
        <!-- MAVROS -->
        <!--<arg name="fcu_url" default="udp://:14542@localhost:14582"/>
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <arg name="tgt_component" value="1"/>
        </include>-->
        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find px4_application)/config/mavros_config/px4_pluginlists.yaml" />
            <arg name="config_yaml" value="$(find px4_application)/config/mavros_config/px4_config_2.yaml" />
            <arg name="fcu_url" value="udp://:14542@localhost:14582" />
            <arg name="gcs_url" value="" />
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))" />
            <arg name="tgt_component" value="1" />
	    </include>

        <!--<node pkg="tf" type="static_transform_publisher" name="tf_depth_camera"
            args="0 0 0 -1.57 0 -1.57 base_link iris_2/depth_camera_link 100"/>-->
    </group>
	<!-- run the rviz -->
	<!-- <arg name="visualization" default="true"/>
	<group if="$(arg visualization)">
		<node type="rviz" name="rviz" pkg="rviz" args="-d $(find px4_application)/config/rviz_config/trajectory_config.rviz" />
    </group> -->

</launch>
<!-- to add more UAVs (up to 10):
Increase the id
Change the name space
Set the FCU to default="udp://:14540+id@localhost:14550+id"
Set the malink_udp_port to 14560+id) -->
